src/pybind/mgr/dashboard/frontend/src/app/ceph/new-dashboard/dashboard/dashboard.component.ts

```diff
@@ -124,4 +139,138 @@ export class DashboardComponent implements OnInit, OnDestroy {
       });
     });
   }
+
+  getPrometheusData(selectedTime: any) {
+    if (this.timerGetPrometheusDataSub) {
+      this.timerGetPrometheusDataSub.unsubscribe();
+    }
+    this.timerGetPrometheusDataSub = timer(0, this.timerTime).subscribe(() => {
+      selectedTime = this.updateTimeStamp(selectedTime);
+      this.prometheusService
+        .getPrometheusData({
+          params: Promqls.USEDCAPACITY,
+          start: selectedTime['start'],
+          end: selectedTime['end'],
+          step: selectedTime['step']
+        })
+        .subscribe((data: any) => {
+          if (data.result.length) {
+            this.usedCapacity = data.result[0].values;
+          }
+        });
+
+      this.prometheusService
+        .getPrometheusData({
+          params: Promqls.IPS,
+          start: selectedTime['start'],
+          end: selectedTime['end'],
+          step: selectedTime['step']
+        })
+        .subscribe((data: any) => {
+          if (data.result.length) {
+            this.ips = data.result[0].values;
+          }
+        });
+
+      this.prometheusService
+        .getPrometheusData({
+          params: Promqls.OPS,
```

Comment:
I see a lot of repetition
```suggestion
  for (const [query_name, query] of queries) {
      this.prometheusService
        .getPrometheusData({
          params: query,
          start: selectedTime['start'],
          end: selectedTime['end'],
          step: selectedTime['step']
        })
        .subscribe((data: any) => {
          if (data.result.length) {
            this.queryResults[query_name] = data.result[0].values;
          }
        });

```